api/envoy/api/v2/cds.proto

```diff
@@ -157,6 +158,21 @@ message Cluster {
   // then hosts is required.
   repeated core.Address hosts = 7;
 
+  // Setting this is required for specifying members of
+  // :ref:`STATIC<envoy_api_enum_value_Cluster.DiscoveryType.STATIC>`,
+  // :ref:`STRICT_DNS<envoy_api_enum_value_Cluster.DiscoveryType.STRICT_DNS>`
+  // or :ref:`LOGICAL_DNS<envoy_api_enum_value_Cluster.DiscoveryType.LOGICAL_DNS>` clusters.
+  // This field supersedes :ref:`hosts<envoy_api_field_Cluster.hosts>` field.
+  //
+  // .. attention::
+  //
+  //   Setting this allows CDS static/DNS assignments to contain embedded EDS equivalent
+  //   :ref:`endpoint assignments<envoy_api_msg_ClusterLoadAssignment>`.
+  //   Setting this overrides :ref:`hosts<envoy_api_field_Cluster.hosts>` values.
```

Comment:
@dio can you make a note to deprecate the hosts field and add to deprecated.md when you do the PR implementing this?

Reply:
Sure, 2ab4f13. Thanks for reminding me.